lib/exq_scheduler/scheduler/server.ex

```diff
@@ -41,26 +43,49 @@ defmodule ExqScheduler.Scheduler.Server do
 
   def init(env) do
     storage_opts = Storage.build_opts(env)
-
-    _ = Storage.load_schedules_config(storage_opts, env)
+    schedules = Storage.load_schedules_config(env)
 
     state = %State{
-      schedules: Storage.get_schedules(storage_opts),
+      schedules: schedules,
       storage_opts: storage_opts,
       server_opts: build_opts(env),
-      env: env
+      env: env,
+      start_time: Time.now()
     }
 
-    Enum.filter(state.schedules, &Storage.is_schedule_enabled?(storage_opts, &1))
-    |> Storage.persist_schedule_times(state.storage_opts)
-
-    next_tick(self(), 0)
+    update_initial_schedule_times(:first, state)
     {:ok, state}
   end
 
+  def update_initial_schedule_times(:first, state) do
+    storage_opts = state.storage_opts
+    if Storage.storage_connected?(storage_opts) do
+      Enum.filter(state.schedules, &Storage.is_schedule_enabled?(storage_opts, &1))
+      |> Enum.filter(fn schedule ->
+        Storage.get_schedule_first_run_time(storage_opts, schedule) == nil
+      end)
+      |> Storage.persist_schedule_times(storage_opts, state.start_time)
+
+      Enum.map(state.schedules, fn schedule ->
+        Storage.persist_schedule(schedule, storage_opts)
+      end)
+
+      next_tick(self(), 0)
+    else
+      Process.send_after(self(), :first, @storage_reconnect_timeout)
+    end
+  end
+
   def handle_info({:tick, time}, state) do
-    handle_tick(state, time)
-    next_tick(self(), state.server_opts.timeout)
+    timeout =
+    if not Storage.storage_connected?(state.storage_opts) do
```

Comment:
can you use `!`. not/and/or has slightly different semantics than `&&/||/!`. I prefer to use the later everywhere.